src/main/scala/com/etsy/conjecture/scalding/util/HyperparameterSearcher.scala

```diff
@@ -0,0 +1,186 @@
+package com.etsy.conjecture.scalding.util
+
+import java.io.Serializable
+
+import cascading.pipe.Pipe
+import com.etsy.conjecture.data._
+import com.etsy.conjecture.evaluation._
+import com.etsy.conjecture.model._
+import com.etsy.conjecture.scalding.evaluate.{BinaryEvaluator, GenericEvaluator, MulticlassEvaluator, RegressionEvaluator}
+import com.etsy.conjecture.scalding.train._
+import com.twitter.scalding.Dsl._
+import com.twitter.scalding._
+
+import scala.util.Random
+
+/**
+  * Samples random parameter values to perform a fast efficient hyperparameter search
+  */
+abstract class HyperparameterSearcher[L <: Label, M <: UpdateableModel[L, M], E <: ModelEvaluation[L]]
```

Comment:
Yea sorry about that haha I wanted it to be able to be used for all types of labels and models.  When I open the review branch for the BigData side it'll be easier to see how it comes into play.